zip-0316.rst

```diff
@@ -592,14 +631,29 @@ to a single Address with a given Typecode (and at most one Transparent
 Address) means that this is also the maximum length as of NU5 activation.
 
 For longer UAs (when other Typecodes are added), the cost increases to 6
-BLAKE2b compressions for :math:`128 < \ell_M \leq 192`, and 10 BLAKE2b
-compressions for :math:`192 < \ell_M \leq 256`, for example. The maximum
-cost for which the algorithm is defined would be 768 BLAKE2b compressions
-at :math:`\ell_M = 16448` bytes. We will almost certainly never add enough
-Typecodes to reach that, and we might want to define a smaller limit.
+BLAKE2b compressions for :math:`128 < \ell_M \leq 192,` and 10 BLAKE2b
+compressions for :math:`192 < \ell_M \leq 256,` for example. The maximum
+cost for which the algorithm is defined would be 196608 BLAKE2b compressions
+at :math:`\ell_M = 4194368` bytes.
+
+A naïve implementation of the :math:`F4Jumble^{-1}` function would require
+roughly :math:`\ell_M` bytes plus the size of a BLAKE2b hash state. However,
+it is possible to reduce this by streaming the :math:`d` part of the jumbled
+encoding three times from a less memory-constrained device. It is essential
+that the streamed value of :math:`d` is the same on each pass, which can be
+verified using a MAC (with key held only by the Consumer) or
+collision-resistant hash function. After the first pass of :math:`d`, the
+implementation is able to compute :math:`y`; after the second pass it is
+able to compute :math:`a`; and the third allows it to compute and
+incrementally parse :math:`b.` The maximum memory usage during this process
+would be 128 bytes (for the cached :math:`c` or :math:`y`), plus two BLAKE2b
+hash states.
+
+Since this streaming implementation of :math:`F4Jumble^{-1}` is quite
+complicated, we do not require all Consumers to support it. If a Consumer
+implementation cannot support UAs / UVKs up to the maximum length, it MUST
+nevertheless support UAs / UVKs with :math:`\ell_M` of at least 256 bytes.
```

Comment:
Having two maximum lengths supported indicates two conformance levels
for UAs/UVKs: minimum for non-experimental types and the maximum including 
experimental types.

Reply:
I will clarify this.